lib/koa-authenticator.js

```diff
@@ -13,11 +14,15 @@ module.exports = function(escherConfig = {}, logger) {
         yield next;
       }
     } catch (ex) {
-      if (logger && logger.error) {
-        logger.error('authentication_request_error', ex.message, ex);
+      if (ex instanceof AuthenticationError) {
+        if (logger && logger.error) {
+          logger.error('authentication_request_error', ex.message, ex);
+        }
+
+        this.throw(401, ex.message);
       }
 
-      this.throw(401, ex.message);
+      this.throw(ex);
```

Comment:
I think error handling should be done on a different level, we should just rethrow the original error here